### PR TITLE
Document OS limits

### PR DESCRIPTION
**Pull request by pentschev**
This PR adds a new RTD section to document some known issues that may arise due to OS limits.

### CODE REVIEW

File: docs/source/os-limits.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+File Descriptors
+----------------
+
+In sockets-based connections, multiple file descriptors may be open to establish connections between endpoints. When UCX is establishing connection between endpoints via protocols such as TCP, an error such as below may occur:
+
+::
+
+    ucp.exceptions.UCXError: User-defined limit was reached
+
+One possible cause for this is that the limit establish by the OS or system administrators has been reached by the user. This limit can be checked with:
```
Comment (quasiben) [inline docs/source/os-limits.rst:16]:
```suggestion
One possible cause for this is that the limit established by the OS or system administrators has been reached by the user. This limit can be checked with:
```

File: docs/source/os-limits.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+::
+    $ ulimit -n
+
+If the user has permission to do so, this limit can be increased by typing the new limit after the command above. For example, to set a new limit of 1 million, the following should be executed:
```
Comment (quasiben) [inline docs/source/os-limits.rst:21]:
```suggestion
If the user has permission to do so, the file descriptor limit can be increased by typing the new limit after the command above. For example, to set a new limit of 1 million, the following should be executed:
```

**Review by quasiben — COMMENTED** [review]
Thank you for writing this up @pentschev . Left a few small suggestions

**Comment by pentschev** [comment]
Thanks @quasiben , suggestions applied!

**Comment by pentschev** [comment]
@quasiben could you approve/merge if you agree with the current text?

**Review by quasiben — APPROVED** [review]
Thanks @pentschev

**Comment by quasiben** [comment]
gpuCI merge

**Comment by quasiben** [comment]
@gpucibot merge

**Pull request merged**